src/cobo/handshake.c

```diff
@@ -410,13 +465,66 @@ static int encode_addr(struct sockaddr *addr, unsigned char *target_addr, uint16
    return 0;
 }
 
+int random_number(uint64_t *unique_value) {
+   int local_errno;
+   uint64_t rand_num, uuid_num;
+
+   //random number file
+   FILE *random_file = fopen("/dev/random", "r");
+   if (!random_file) {
+      local_errno = errno;   
+      debug_printf("Failed to open /dev/random. Error: %s\n",strerror(local_errno));
+      fclose(random_file);
+      return -1;
+   }
+   
+   if (fread(&rand_num, sizeof(rand_num), 1, random_file) != 1) {  
+      local_errno = errno;   
+      debug_printf("Failed to read random number. Error: %s\n",strerror(local_errno)); 
+      fclose(random_file);
+      return -1;
+   }
+   fclose(random_file);
+
+   //uuid file
+   FILE *uuid_file = fopen("/proc/sys/kernel/random/uuid", "r");
+   if (!uuid_file) {
+      local_errno = errno;   
+      debug_printf("Failed to open /proc/sys/kernel/random/uuid. Error: %s\n",strerror(local_errno));
+      fclose(uuid_file);
+      return -1;
+   }
+   if (fread(&uuid_num, sizeof(uuid_num), 1, uuid_file) != 1) {  
+      local_errno = errno;   
+      debug_printf("Failed to read uuid. Error: %s\n",strerror(local_errno)); 
+      fclose(uuid_file);
+      return -1;
+   }
+   fclose(uuid_file);
+
+   time_t now = time(NULL);
+
+   // Combine timestamp,random number, uuid
+   *unique_value = now ^ rand_num ^ uuid_num;
```

Comment:
We don't want to xor the data together. We're essentially xor'ing a random number with a partially-random number with a non-random number. I don't think that produces anything interesting. The goal is to make a unique number. I think we could do that by concat'ing into 128 bits (or fill in multiple 64-bit fields) with a random number, session id, and a counter.